### ISSUE TITLE
Refreshing tasks - Failed to notify build listener - Found 0 tasks

### ISSUE THREAD

**Issue by hknoerrer** [badsyntax/vscode-gradle#1225]
Extension Name: vscode-gradle
Extension Version: 3.12.0
OS Version: OSX 12.2.1
VSCode version: 1.67.2

**Describe the bug**

**To Reproduce**
mkdir gradle4-demo && cd gradle4-demo
gradle init
code .
Check output of "Gradle for Java"

**Expected behavior**
No thrown error during startup.
After applying plugins in build.gradle the list of tasks should contain the additional tasks

**Screenshots**

**Output from "Gradle for Java"**

```
[debug] [gradle-server] Starting server
[debug] [gradle-server] Gradle Server cmd: /Users/XXXX/.vscode/extensions/vscjava.vscode-gradle-3.12.0/lib/gradle-server 54934
[info] [gradle-server] Server started, listening on 54934
[info] Gradle client connected to server
[debug] Refreshing tasks
[info] Java Home: /Users/XXXX/.asdf/installs/java/corretto-11.0.15.9.1
[info] JVM Args: --add-opens,java.base/java.util=ALL-UNNAMED,-XX:+HeapDumpOnOutOfMemoryError,-Xmx1024m,-Dfile.encoding=UTF-8,-Duser.country=DE,-Duser.language=de,-Duser.variant
[info] Gradle User Home: /Users/XXXX/.gradle
[info] Gradle Version: 4.10.3
[info] CONFIGURE SUCCESSFUL in 1s

FAILURE: Build failed with an exception.

* What went wrong:
Failed to notify build listener.
> 'java.lang.Class org.gradle.api.reflect.TypeOf.getConcreteClass()'

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
[error] Error getting build for /Users/XXXX/entwicklung/otherSources/gradle4-demo: Could not run build action using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-4.10.3-bin.zip'.
[error] [gradle-server] Could not run build action using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-4.10.3-bin.zip'.
[info] Java Home: /Users/XXXX/.asdf/installs/java/corretto-11.0.15.9.1
[info] JVM Args: --add-opens,java.base/java.util=ALL-UNNAMED,-XX:+HeapDumpOnOutOfMemoryError,-Xmx1024m,-Dfile.encoding=UTF-8,-Duser.country=DE,-Duser.language=de,-Duser.variant
[info] Gradle User Home: /Users/XXXX/.gradle
[info] Gradle Version: 4.10.3
[info] CONFIGURE SUCCESSFUL in 0s

FAILURE: Build failed with an exception.

* What went wrong:
Failed to notify build listener.
> 'java.lang.Class org.gradle.api.reflect.TypeOf.getConcreteClass()'

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

[error] [gradle-server] Could not run build action using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-4.10.3-bin.zip'.
BUILD FAILED in 0s
[error] Error getting build for /Users/XXXX/entwicklung/otherSources/gradle4-demo: Could not run build action using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-4.10.3-bin.zip'.
[info] Found 0 tasks

```

**Does the bug still exist if you disable all other extensions?**
Just enabled "Gradle Language Support - v0.2.3" / "Language Support for Java(TM by Red Hat - v.1.6.0" / "German Language Pack v.1.67.2"

**Additional context**
The problem occurs in company network (with proxy settings) and outside (without proxy settings)

**Comment by hknoerrer** [comment]
BTW: same steps with Gradle Version: 7.1.1 -> no error / additional task were shown.

**Comment by macjabeth** [comment]
I'm getting a similar issue with my project:

```
[info] [gradle-server] Server started, listening on 60644
[info] Gradle client connected to server
[info] Java Home: /Library/Java/JavaVirtualMachines/temurin-8.jdk/Contents/Home
[info] JVM Args: -XX:+HeapDumpOnOutOfMemoryError,-Xmx1024m,-Dfile.encoding=UTF-8,-Duser.country=US,-Duser.language=en,-Duser.variant
[info] Gradle User Home: /Users/xxx####/.gradle
[info] Gradle Version: 4.7
[info] CONFIGURE SUCCESSFUL in 6s
[error] FAILURE: Build failed with an exception.

* What went wrong:
Failed to notify build listener.
> org.gradle.api.plugins.ExtensionsSchema.getElements()Ljava/lang/Iterable;

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 6s
[error] Error getting build for /Users/xxx####/Code/Compliance_ecpmmt-rest-api: Could not run build action using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-4.7-bin.zip'.
[info] Found 0 tasks
[error] [gradle-server] Could not run build action using connection to Gradle distribution 'https://services.gradle.org/distributions/gradle-4.7-bin.zip'.
[info] Build file opened: /Users/xxx####/Code/Compliance_ecpmmt-rest-api/build.gradle
```

**Comment by CsCherrYY** [comment]
@hknoerrer  I find the root cause - we're using the Gradle API to get the task info, which is available since Gradle 5.0. See: https://docs.gradle.org/current/javadoc/org/gradle/api/reflect/TypeOf.html#getConcreteClass--, So we'd better have a guard for it.

**Comment by CsCherrYY** [comment]
@macjabeth could you please provide a sample project to help me reproduce this issue? I'm not able to get the correct cause from your log. Looks like something is wrong with `ExtensionsSchema`.
